Migrate to Pluto

NitishGadangi opened the pull request:
fix to [#133](https://github.com/JuliaEarth/GeoStats.jl/issues/133)  (previously #5)

**Description**
- Migration of Tutorials from Jupyter Notebook to Pluto.jl Notebook

**Remarks**
- Check Deliverables for Progress
- Not yet deployed on http://pluto-on-binder.glitch.me/ (But works flawlessly on Local Machine)
- Interact.jl -> PlutoUI.jl

**Instructions**
Go to the Project Directory
- Run Julia and activate the environment:
``` shell
julia> ]
(v1.5) pkg> activate .
(SomeProject) pkg> instantiate

julia> import Pluto
julia> Pluto.run()
```
- Open the respective tutorial on Pluto

**Deliverables:**
- [x] Estimation problems
- [x] Variogram modeling
- [x] Anisotropic models
- [x] Directional variograms
- [x] Declustered statistics
- [x] Two-point statistics
- [x] Gaussian processes
- [x] Image quilting
- [x] Cookie-cutter

- [ ] Make the Notebooks deployable on http://pluto-on-binder.glitch.me/
- [x] Update Readme with Required Instructions

NitishGadangi committed:
create Environment and Add Packages

NitishGadangi committed:
add EstimationProblems.jl

Pluto.jl Notebook for EstimationProblems

NitishGadangi committed:
add VariogramModeling.jl

Pluto.jl Notebook for VariogramModeling
Removed Interact.jl and added PlutoUI.jl

juliohm commented on notebooks/pluto/Project.toml:
```diff
@@ -0,0 +1,6 @@
+GeoStats = "dcc97b0b-8ce5-5539-9008-bb190f959ef6"
+GeoStatsImages = "7cd16168-b42c-5e7d-a585-4f59d326662d"
+Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
+Pluto = "c3e4b0f8-55cb-11ea-2926-15256bba5781"
```
I think we don't need Pluto as a dependency, just PlutoUI, can you double check?

NitishGadangi replied:
I thought it will be more intuitive to add Pluto too,
as it will be smooth for the User to directly launch the Pluto from the same terminal (right in the same environment)
and he doesn't need to do `] add Pluto` again

juliohm replied:
What I am trying to say is that Pluto isn't a dependency of the tutorials, and people willing to run these notebooks will have Pluto installed globally as opposed to locally here. Appreciate if you can remove the dependency.

NitishGadangi replied:
Sure, I will remove the dependency. 👍

juliohm commented on notebooks/pluto/VariogramModeling.jl:
It looks like we have an extra white space here. Also, did you open an issue on GitHub about the slider type instability? It is not ideal to explicitly type Float64 here.

NitishGadangi replied:
I think I didn't get you about _extra white space_

and Yup I have opened an issue regarding Instability on PlutoUI
Ref: https://github.com/fonsp/PlutoUI.jl/issues/64

juliohm replied:
It seems that the first line contains more than 4 whitespaces. It may be a GitHub viewer issue though.

juliohm reviewed (changes requested):
Thank you for sharing the PR @NitishGadangi. I have a few comments attached.

Could you please comment on why you needed to create a new folder `pluto/data/` with copies of the csv files?

I will try to visualize the notebooks locally soon.

NitishGadangi committed:
remove duplicates of CSV files

NitishGadangi commented:
>
> Could you please comment on why you needed to create a new folder `pluto/data/` with copies of the csv files?
> 
Thanks for pointing out.
I have pushed a commit that removes the CSV file duplicates. 😀

juliohm commented:
Please continue updating the PR as you port more notebooks. I will be happy to provide more feedback as we go. We are also learning the rough corners of Pluto with this work.

juliohm commented:
Another thing that you can do already is remove the `pluto` subfolder and just use the root Project.toml/Manifest.toml.

juliohm commented:
Also, it would be great if you could group all `using Foo` statements in the first cell of the notebooks using a `begin-end` block. There are some old notebooks there that are adding `using` statements in the middle of the story, but I think we can move them up.

NitishGadangi committed:
add Declustered statistics

NitishGadangi committed:
move pluto files to /notebooks

NitishGadangi committed:
add AnisotropicModels.jl

build GR package to fix issue with Plots.jl

error compiling _plot!: error compiling _display: could not load library "libGR.dll"

NitishGadangi committed:
add TwoPointStatistics.jl

used ``Latex`` in few places of markdown instead of $$

NitishGadangi committed:
add GaussianProcesses.jl

used PlutoUI instead of Interact.jl

NitishGadangi committed:
add CookieCutter.jl

NitishGadangi committed:
add ParallelSimulation.jl

used `if myid() == 1 && nworkers() == 1` to restrict un usual addition of processes (addprocs) due to repeated cell execution

NitishGadangi committed:
add ImageQuilting.jl

updating ImageQuilting package has solved the issue with solver packages

NitishGadangi committed:
add DirectionalVariograms.jl

juliohm committed:
Adjustments to Pluto notebooks

juliohm commented:
@NitishGadangi could you please enable edits on this PR so that I can push some adjustments?

I've unmarked some of the notebooks as incomplete because the solve issue is still present. Also, the AnisotropicVariograms.jl notebook has a LaTeX issue that we need to address.

juliohm committed:
Remove ParallelSimulation

juliohm committed:
Fix LaTeX in AnisotropicModels.jl

juliohm committed:
Update notebooks for latest GeoStats.jl

juliohm committed:
Final fix for all notebooks

juliohm commented:
@NitishGadangi all notebooks are working fine now. We are almost there!

Can you please check how we can deploy the notebooks on binder?

juliohm committed:
Update README

juliohm committed:
Remove Jupyter notebooks

NitishGadangi committed:
remove nbviewer links from README

juliohm committed:
Small fixes in README

juliohm committed:
Update README

juliohm commented:
Thanks again for this! It is nice to replace Jupyter by Pluto! The notebooks look much nicer and more interactive than ever! :balloon: :fireworks:
